src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -174,11 +275,40 @@ class BezierCurve
   /// Retrieves the control point at index \a idx
   const PointType& operator[](int idx) const { return m_controlPoints[idx]; }
 
+  /*!
+   * \brief Get a specific weight
+   *
+   * \param [in] idx The index of the weight
+   * \pre Requires that the curve be rational
+   */
+  const T& getWeight(int idx) const
+  {
+    SLIC_ASSERT(isRational());
+    return m_weights[idx];
+  }
+
+  /*!
+   * \brief Set the weight at a specific index
+   *
+   * \param [in] idx The index of the weight
+   * \param [in] weight The updated value of the weight
+   * \pre Requires that the curve be rational
+   * \pre Requires that the weight be positive
+   */
+  void setWeight(int idx, T weight)
+  {
+    SLIC_ASSERT(isRational());
+    SLIC_ASSERT(weight > 0);
+
+    m_weights[idx] = weight;
+  };
+
   /// Checks equality of two Bezier Curve
   friend inline bool operator==(const BezierCurve<T, NDIMS>& lhs,
                                 const BezierCurve<T, NDIMS>& rhs)
   {
-    return lhs.m_controlPoints == rhs.m_controlPoints;
+    return (lhs.m_controlPoints == rhs.m_controlPoints) &&
+      (lhs.m_weights == rhs.m_weights);
```

Comment:
Note -- this implementation would return false when comparing a nonrational curve and a rational curve whose weights are all 1? 

I think this is the right call, but wanted to highlight it.

Reply:
This is what I had in mind, yes. I interpreted it this way since it seems like, without any tolerances on the control nodes, this is more a check of if they are the same C++ object, rather than the same mathematical object.